### PR TITLE
Codegen: Add TVMCG_DEBUG_ALLOCATIONS and handle error code properly

### PR DESCRIPTION
**Pull request by PhilippvK**

### CODE REVIEW

**Comment by PhilippvK** [comment]
Waiting to CI to complete before merging...

**Comment by PhilippvK** [comment]
okay, i will update the tvm patches later to fix this...

File: src/CodeGen.cpp [hunk 1]
```diff
@@ -161,11 +161,23 @@ size_t TVMWrap_GetNumOutputs()
 void* TVMBackendAllocWorkspace(int device_type, int device_id, uint64_t nbytes, int dtype_code_hint,
                                int dtype_bits_hint) {
   void *out_ptr = NULL;
-  tvm_crt_error_t err = StackMemoryManager_Allocate(&g_workspace, nbytes, &out_ptr);
+#ifdef TVMCG_DEBUG_ALLOCATIONS
+  if (nbytes > (g_workspace.workspace+g_workspace.workspace_size-g_workspace.next_alloc)) {
```
Comment (rafzi) [inline src/CodeGen.cpp]: the missing spaces would not conform to the clang-format

**Review by rafzi — APPROVED** [review]
the define could be enabled by default on the debug build

**Comment by PhilippvK** [comment]
I have added some linter checks to the CI config to ensure that the formatting of c++ is fine. (Copied most parts over from another project, so it was not too much effort) There are still some issues with a few targets I will fix later…

**Comment by PhilippvK** [comment]
Will close this and open another PR when I have more time.

**Pull request closed without merging**